chain/chain/src/store.rs

```diff
@@ -2081,6 +2043,144 @@ impl<'a> ChainStoreUpdate<'a> {
+            }
+            DBCol::ColChunkPerHeightShard => {
+                store_update.delete(col, key);
+                self.chain_store.chunk_hash_per_height_shard.cache_remove(key);
+            }
+            DBCol::ColNextBlockWithNewChunk => {
+                store_update.delete(col, key);
+                self.chain_store.next_block_with_new_chunk.cache_remove(key);
+            }
+            DBCol::ColStateHeaders => {
+                store_update.delete(col, key);
+            }
+            DBCol::ColBlock => {
+                store_update.delete(col, key);
+                self.chain_store.blocks.cache_remove(key);
+            }
+            DBCol::ColBlockExtra => {
+                store_update.delete(col, key);
+                self.chain_store.block_extras.cache_remove(key);
+            }
+            DBCol::ColNextBlockHashes => {
+                store_update.delete(col, key);
+                self.chain_store.next_block_hashes.cache_remove(key);
+            }
+            DBCol::ColChallengedBlocks => {
+                store_update.delete(col, key);
+            }
+            DBCol::ColBlocksToCatchup => {
+                store_update.delete(col, key);
+            }
+            DBCol::ColStateChanges => {
+                store_update.delete(col, key);
+            }
+            DBCol::ColBlockRefCount => {
+                store_update.delete(col, key);
+                self.chain_store.block_refcounts.cache_remove(key);
+            }
+            DBCol::ColReceiptIdToShardId => {
+                store_update.delete(col, key);
+                self.chain_store.receipt_id_to_shard_id.cache_remove(key);
+            }
+            DBCol::ColTransactions => {
+                store_update.delete(col, key);
+                self.chain_store.transactions.cache_remove(key);
+            }
+            DBCol::ColChunks => {
+                store_update.delete(col, key);
+                self.chain_store.chunks.cache_remove(key);
+            }
+            DBCol::ColChunkExtra => {
+                store_update.delete(col, key);
+                self.chain_store.chunk_extras.cache_remove(key);
+            }
+            DBCol::ColPartialChunks => {
+                store_update.delete(col, key);
+                self.chain_store.partial_chunks.cache_remove(key);
+            }
+            DBCol::ColInvalidChunks => {
+                store_update.delete(col, key);
+                self.chain_store.invalid_chunks.cache_remove(key);
+            }
+            DBCol::ColChunkHashesByHeight => {
+                store_update.delete(col, key);
+                self.chain_store.chunk_hash_per_height_shard.cache_remove(key);
+            }
+            DBCol::ColStateParts => {
+                store_update.delete(col, key);
+            }
+            DBCol::ColState => {
+                store_update.delete(col, key);
+            }
+            DBCol::ColTrieChanges => {}
```

Comment:
store_update.delete(col, key);

Reply:
This is no-op, just to bump gc count. real gc happened by `store().get_ser(ColTrieChanges, ...)..map(|tries| tries.revert_insertions(&trie_changes)`
or `tries.delete_insertions`

We can move gc logic of tries column to gc method too, but that has to take a parameter of current GCMode, make it incompatible with gc_col method. We can make a gc_col_trie_changes similar to gc_col_block_per_height, do you think it's better?

Reply:
There are two different columns to GC: ColState and ColTrieChanges.
ColState is removed inside `revert_insertions` or `delete_insertions`. ColTrieChanges is the column that should be cleared manually later.

Technically, the data in ColTrieChanges is what should be cleared in ColState. After clearing ColState, we don't need a record in ColTrieChanges anymore and remove it in `.map(|_| { ...`. ColTrieChanges is not dependent on GCMode and that's why I suggest to execute it here.

Reply:
Got it